docs/ru/manuals/addressing.md

```diff
@@ -0,0 +1,246 @@
+
+![bean](images/addressing/team_editor.png)
+
+При такой структуре игровой объект "bean" все также может ссылаться на компонент "controller" в объекте "buddy" по адресу `"buddy#controller"`.
+
+![bean](images/addressing/collection_team.png)
+
+Если добавить второй экземпляр "team.collection" (с именем "team_2"), то код, запущенный внутри скрипта экземпляра коллекции "team_2", также будет успешно работать. Экземпляр игрового объекта "bean" из коллекции "team_2" по-прежнему может обращаться к компоненту "controller" в объекте "buddy" по адресу `"buddy#controller"`.
+
+![bean](images/addressing/teams_editor.png)
+
+## Относительная адресация
+
+Адрес `"buddy#controller"` работает для игровых объектов в обеих коллекциях, потому что это *относительный* адрес. Каждая из коллекций, "team_1" и "team_2", задает новый контекст именования, или, если хотите, "пространство имён". Defold позволяет избежать конфликтов при именовании, учитывая контекст, создаваемый коллекцией для адресации:
+
+![relative id](images/addressing/relative_same.png)
+
+- В контексте именования коллекции "team_1" однозначно идентифицируются игровые объекты "bean" и "buddy".
+- Аналогичным образом, в контексте именования коллекции "team_2", игровые объекты "bean" и "buddy" также имеют уникальную идентификацию.
+
+Относительная адресация работает путем автоматического добавления текущего контекста именования при определении целевого адреса. Это опять-таки чрезвычайно удобно, поскольку появляется возможность создавать группы игровых объектов с кодом и эффективно использовать их на всех этапах разработки игры.
+
+### Сокращения
+
+Defold предлагает два полезных сокращения, которые можно использовать для отправки сообщений без указания полного URL:
+
+:[Shorthands](../shared/url-shorthands.md)
+
+## Пути игровых объектов
+
+Чтобы правильно понять принцип именования, посмотрим, что происходит при сборке и запуске проекта:
+
+1. Редактор считывает загрузочную коллекцию ("main.collection") и все ее содержимое (игровые объекты и другие коллекции).
+2. Для каждого статичного игрового объекта компилятор создает идентификатор. Они формируются в виде "путей", начинающихся от загрузочной коллекции, далее по иерархии коллекции к объекту. На каждом уровне добавляется символ '/'.
+
+В приведенном выше примере игра будет работать со следующими 4 игровыми объектами:
+
+- /team_1/bean
+- /team_1/buddy
+- /team_2/bean
+- /team_2/buddy
+
+::: sidenote
+Идентификаторы хранятся в виде хэшированных значений. В процессе выполнения также сохраняется хэш-состояние для каждого идентификатора коллекции, которое используется для последующего хэширования строки относительного адреса к абсолютному идентификатору.
```

Comment:
ох, какое сложное построение. довольно тяжело понять.

Hey @britzl could you help us a bit pls.
This sentence is pretty difficult for understanding (esp. the second part):

> The runtime also stores the hash state for each collection identity which is used to continue hashing relative string to an absolute id.

I'm not sure I fully understand that, so it's hard to translate correctly. Could you please explain what it means?
Thank you!

Reply:
Lol! Yes, that is not a very friendly sentence :-)

"For each static game object, the compiler creates an identifier. These are built as “paths” starting at the bootstrap root, down the collection hierarchy to the object. A ‘/’ character is added at each level.
For our example above, the game will run with the following 4 game objects:

/team_1/bean
/team_1/buddy
/team_2/bean
/team_2/buddy"

So the above part is clear. Each game object is given a unique id that is created as a "path" including the ids of the collections that makes up the full hierarchy down to the individual game objects.

But what that sidenote is actually about is not clear to me either. And in all honesty I don't see how it would help the developer either. I propose that it is removed from the english text as well.